frontend/src/utils/index.js

```diff
@@ -19,3 +20,22 @@ export const slugifiedNameFromFiles = files => {
  * @returns {Promise<Promise<Blob>>}
  */
 export const b64toBlob = base64 => fetch(base64).then(res => res.blob())
+
+/**
+ * Get the repo name from its url
+ * @param url
+ * @returns {string}
+ */
+export const urlToName = url => {
+  url = new URL(url)
+  return path.basename(url.pathname, path.extname(url.pathname))
+}
+
+/**
+ * Get the project name from the `path` object in `next.router`.
+ * @param path
+ * @returns {*}
+ */
+export const projectNameFromPath = path => {
+  return path.split('/').slice(3).join('/')
+}
```

Comment:
Would be good to have an example in each of these, something like this (if it's correct);
```js
/*

e.g. /user/project/x/y -> /user/project 

*/
```